StatPlugin.cs

```diff
@@ -1,50 +1,62 @@
 using System;
 using System.Collections.Generic;
 using System.Text;
+using System.Text.RegularExpressions;
 
 namespace DNWS
 {
-  class StatPlugin : IPlugin
-  {
-    protected static Dictionary<String, int> statDictionary = null;
-    public StatPlugin()
-    {
-      if (statDictionary == null)
-      {
-        statDictionary = new Dictionary<String, int>();
+	class StatPlugin : IPlugin
+	{
+		protected static Dictionary<String, int> statDictionary = null;
+		public StatPlugin()
+		{
+			if (statDictionary == null)
+			{
+				statDictionary = new Dictionary<String, int>();
 
-      }
-    }
+			}
+		}
 
-    public void PreProcessing(HTTPRequest request)
-    {
-      if (statDictionary.ContainsKey(request.Url))
-      {
-        statDictionary[request.Url] = (int)statDictionary[request.Url] + 1;
-      }
-      else
-      {
-        statDictionary[request.Url] = 1;
-      }
-    }
-    public HTTPResponse GetResponse(HTTPRequest request)
-    {
-      HTTPResponse response = null;
-      StringBuilder sb = new StringBuilder();
-      sb.Append("<html><body><h1>Stat:</h1>");
-      foreach (KeyValuePair<String, int> entry in statDictionary)
-      {
-        sb.Append(entry.Key + ": " + entry.Value.ToString() + "<br />");
-      }
-      sb.Append("</body></html>");
-      response = new HTTPResponse(200);
-      response.body = Encoding.UTF8.GetBytes(sb.ToString());
-      return response;
-    }
+		public void PreProcessing(HTTPRequest request)
+		{
+			if (statDictionary.ContainsKey(request.Url))
+			{
+				statDictionary[request.Url] = (int)statDictionary[request.Url] + 1;
+			}
+			else
+			{
+				statDictionary[request.Url] = 1;
+			}
+		}
+		public HTTPResponse GetResponse(HTTPRequest request)
+		{
+			//Console.WriteLine(string.Join(", ", request.whole_request));
+			//Console.WriteLine(string.Join(", ", request.getPropertyByKey("Accept-Language")));
 
-    public HTTPResponse PostProcessing(HTTPResponse response)
-    {
-      throw new NotImplementedException();
-    }
-  }
+			HTTPResponse response = null;
```

Comment:
The instruction said that you should do this in separate plugin, e.g,., ClientInfo plugin, not do it in stat.